### ISSUE TITLE
"Not ready for alien host"

### ISSUE THREAD

**Issue by eserte** [jhthorsen/mojolicious-plugin-assetpack#43]
Most tests are skipped on non-linux systems. But it seems that running the test suite on FreeBSD systems is no worse than on Linux systems --- all tests but the sass.t test pass, but this one also fails on my Linux system (maybe due to unexpected output from the CSS::Sass --- see http://matrix.cpantesters.org/?dist=Mojolicious-Plugin-AssetPack%200.41 )

**Comment by jhthorsen** [comment]
I don't get the error message: It says "Cannot execute 'sass'..." which comes from https://github.com/jhthorsen/mojolicious-plugin-assetpack/blob/0.41/lib/Mojolicious/Plugin/AssetPack/Preprocessor.pm#L94, but the only way to get to there (from what I can undestand) is by https://github.com/jhthorsen/mojolicious-plugin-assetpack/blob/0.41/lib/Mojolicious/Plugin/AssetPack/Preprocessor/Sass.pm#L52

That doesn't have anything to do with CSS::Sass. Could it be that a5bdb275bdbafb720a1d54fc55cf09a095e7060c fixed this..?

**Comment by eserte** [comment]
Possinly, test failures stopped with 0.43.

And with the following change tests still pass on my freebsd system (though I don't know if it would work on more exotic systems like MSWin32):
```
diff --git i/t/Helper.pm w/t/Helper.pm
index 28f422c..2a74f0f 100644
--- i/t/Helper.pm
+++ w/t/Helper.pm
@@ -21,7 +21,7 @@ sub import {
   my $class  = shift;
   my $caller = caller;
 
-  plan skip_all => 'Not ready for alien host' unless $^O eq 'linux';
+#  plan skip_all => 'Not ready for alien host' unless $^O eq 'linux';
 
   unlink glob 't/public/packed/*' unless $ENV{TEST_KEEP_FILES};
 
```

**Comment by jhthorsen** [comment]
Version 0.46 is on it's way to CPAN, without the "skip_all" rule above.

I wonder how it will turn out this time, but I feel more confident I can get to the bottom of the failing tests with your help.

Thanks for the insight you have provided allready :+1:

**Comment by eserte** [comment]
First impression: http://217.199.168.174/cgi-bin/cpantestersmatrix.pl?dist=Mojolicious-Plugin-AssetPack%200.46
The two failures in the 5.10.1 line are probably systems with Mojolicious 5. And MSWin32 does not look good, as suspected.

**Comment by jhthorsen** [comment]
These are the reports I have:
- MSWin32-x86-multi-thread / 5.16.0:
  - FAIL http://www.cpantesters.org/cpan/report/661a17f9-6bf5-1014-99e3-837b5c3ee46f
- MSWin32-x86-multi-thread-64int / 5.20.2:
  - FAIL http://www.cpantesters.org/cpan/report/fd28bb89-6c10-1014-b211-abc6c4a9a683
- amd64-freebsd / 5.10.1:
  - FAIL http://www.cpantesters.org/cpan/report/0ff0d6e2-d896-11e4-977e-7d7ae0bfc7aa
- i86pc-solaris-thread-multi-64 / 5.16.0:
  - FAIL http://www.cpantesters.org/cpan/report/dba91daa-d8bc-11e4-b221-9e126cbd7f71
- x86_64-linux / 5.10.1:
  - FAIL http://www.cpantesters.org/cpan/report/11423b44-d896-11e4-977e-7d7ae0bfc7aa

Do you have any other?

**Comment by jhthorsen** [comment]
Seems like one of the reasons for so many failing tests is because I depend on an old Mojolicious version. I've bumped and released a new version now. Looking forward to more test results.

**Comment by jhthorsen** [comment]
Can you provide any details on what is going on here? http://www.cpantesters.org/cpan/report/959aef60-d9fb-11e4-b952-c4ccdfbfc7aa

**Comment by eserte** [comment]
Looks like Mojolicious 5.41 is causing the problems here. I can reproduce the same failures on Linux systems if downgrading to this Mojolicious version. All other tester reports list Mojolicious 6.05 (checked with `ctgetreports Mojolicious-Plugin-AssetPack --q mod:Mojolicious`).

**Comment by eserte** [comment]
Checked also with Mojolicious-5.82 --- this version works fine with your module. So the breakage is probably somewhere between 5.41 and 5.82.

**Comment by jhthorsen** [comment]
Thanks! I found the issue: ->tap() does not support arguments before Mojolicious 5.57. Fixed the issue in 7bebc95.

**Comment by jhthorsen** [comment]
I would say this is done now. Thanks for the help!

http://matrix.cpantesters.org/?dist=Mojolicious-Plugin-AssetPack+0.49

Not sure why I had to skip the tests on cygwin though, but I'm in no position to figure it out: 0c738b91b05268f1ea6e0791d6d876cc1c7cc803
